### ISSUE TITLE
Feature Request: Only apply links_space_char to __LinkUrl__ part of Markdown link

### ISSUE THREAD

**Issue by davidsierradz** [vimwiki/vimwiki#1038]
Prior to submitting a new issue make sure to complete these steps:

- [X] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.
- [X] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [X] Include the output of `:VimwikiShowVersion`.

```
:VimwikiShowVersion
Os: Linux                      
Vim: 800                       
Branch: dev                    
Revision: e2a6b82              
Date: 2020-10-19 23:47:10 -0300
```

Hi. With:

```viml
let g:vimwiki_list = [
      \ {'path': '~/notes/',
      \ 'links_space_char': '-',
      \ 'syntax': 'markdown', 'ext': '.md'}]
```

Currently, if I generate a link, this happens:

```
foo bar -> [foo-bar](foo-bar)
```

I would like:

```
foo bar -> [foo bar](foo-bar)
```

The following patch resolves this:

```diff
diff --git a/autoload/vimwiki/base.vim b/autoload/vimwiki/base.vim
index 5c532d2..3ec366b 100644
--- a/autoload/vimwiki/base.vim
+++ b/autoload/vimwiki/base.vim
@@ -2719,7 +2719,7 @@ function! s:normalize_link_syntax_v() abort
   let link = s:safesubstitute(link, '__FileExtension__', file_extension , '')
   " Replace space characters
   let sc = vimwiki#vars#get_wikilocal('links_space_char')
-  let link = substitute(link, '\s', sc, 'g')
+  let link = substitute(link, '(\(.*\))', {m -> substitute(m[0], '\s', sc, 'g')}, 'g')
   " Remove newlines
   let link = substitute(link, '\n', '', '')
```

Maybe we can put this behind a config flag? Also I did't tested it with vimwiki syntax, so I don't know if this will break the setting there or cause problems in other parts of the codebase.

Thanks!

**Comment by tinmarino** [comment]
It is a nice idea. But I dont like flags and prefer opcion like descripcion_handler that takes a string like 'substitute(in, '-', ' ', out)' so other peolple can replace say `/` by `@`.

You see the nice thing with that is that we document that is can break auto reference link and push the responsability to the user hihihi.

@davidsierradz  It would be great if you can submit a PR with this wiki_local option and one or to test if you can.
If you do so, you can earn 1/4 of an hacktobefest T-shirt, yeepee

**Comment by tinmarino** [comment]
Fixed by @vinnyfuria #1132
